src/radical/entk/appman/appmanager.py

```diff
@@ -270,9 +270,28 @@ def run(self):
 
             self._prof.prof('amgr run started', uid=self._uid)
 
+
+            # Submit resource request if not resource allocation done till now or
+            # resubmit a new one if the old one has completed
+            if self._resource_manager:
+                res_alloc_state = self._resource_manager.get_resource_allocation_state()
+                if (not res_alloc_state) or (res_alloc_state in self._resource_manager.get_completed_states()):
+
+                    self._logger.info('Starting resource request submission')
+                    self._prof.prof('init rreq submission', uid=self._uid)
+                    self._resource_manager._submit_resource_request()
+                    res_alloc_state = self._resource_manager.get_resource_allocation_state()
+                    if res_alloc_state in self._resource_manager.get_completed_states():
+                        raise EnTKError(msg="Cannot proceed. Resource allocation ended up in %s"%res_alloc_state)
+
+            else:
```

Comment:
I would suggest to avoid nesting like this:
```
if something:
  do a
  do b
  do c
  if something_else:
    do d
  ...
  do x
else:
  raise(you need something)
```

instead perform the requirement checks all as early as possible, and leave the structure of the actual semantics self contained:

```
# check if we can do stuff
if not something:
  raise(you need something)

# do stuff
do a
do b
do c
if something_else:
  do d
...
do x
```

Reply:
Thanks Andre. I've done this extensively for the wfprocessor and brought down the indentation levels. I'm doing it for the amgr as well, not complete yet.

Reply:
Yep, I saw that! :-D

Reply:
I will check again, but there is only one point in the WFProcessor where I did not want to check for the negation because I think that logic will be harder to process - this is when create a workload of ready tasks or failed tasks.

If there is any other point where you saw scope for this, can you point it please? :)

Reply:
Only here in the app-manager I think.

Reply:
I have used this in all places (but one as mentioned above). The above lines are required to make sure to not proceed/exit if the pilot reaches FAILED (or DONE incorrectly).